backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/document/model/dsl/LuceneIndexSchemaFieldContext.java

```diff
@@ -6,12 +6,28 @@
  */
 package org.hibernate.search.v6poc.backend.lucene.document.model.dsl;
 
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.apache.lucene.index.IndexableField;
 import org.hibernate.search.v6poc.backend.document.model.dsl.IndexSchemaFieldContext;
+import org.hibernate.search.v6poc.backend.document.model.dsl.IndexSchemaFieldTerminalContext;
 
 
 /**
  * @author Guillaume Smet
  */
 public interface LuceneIndexSchemaFieldContext extends IndexSchemaFieldContext {
 
+	/**
+	 * Declares a native field.
+	 *
+	 * @param fieldProducer A bifunction producing a field. The first parameter is the absolute path of the field, the second one is the value.
+	 * @param fieldValueExtractor A function responsible for extracting the value from the field. The parameter is the Lucene field itself.
+	 * @param <V> The type of the value.
+	 * @return The DSL context.
+	 */
+	<V> IndexSchemaFieldTerminalContext<V> asLuceneField(BiFunction<String, V, IndexableField> fieldProducer,
```

Comment:
I expected the field to simply be of type "IndexableField", but I understand you did it this way because users are not always aware of the absolute field path, and we need to give it to them somehow.

However, as it is, the field producer can only create one `IndexableField` per value, which is a bit annoying if one wants to add DocValues, for example. Note I'm not talking about a multi-valued field (which are another matter entirely), but rather about a single value of type `V` leading to multiple `IndexableField` instances, i.e. multiple physical representations (stored, docvalues, index...).

I know the user can always create two fields with different names, and store the docvalues on one side, the indexed value on the other, but I suspect that's not how people expect to use Lucene... Especially if they've been using Hibernate Search 5 and earlier.

One option to fix this would be to use some kind of contributor/collector pattern instead:

```
public interface FieldProducer<V> {
  void produce(String absolutePath, V value, Consumer<IndexableField> collector);
}
```

Another option would be to return a specific type of `IndexFieldAccessor` for this type of fields, one that would expose the field name through a `getAbsolutePath()` method next to `write()`. Then we could make the accessor expect input values of type `IndexableField`, and the user would only have to call `write` multiple times to add multiple `IndexableField` instances to the document.

But maybe you have other ideas?

Reply:
I had an hesitation and thought about the first option too. Will do.